lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -413,14 +268,64 @@ struct AssertOpConversion
         return failure();
       }
     }
+#ifdef USE_ROCM
+    llAssertHIP(op, condition, adaptor.getMessage(), adaptor.getFile(),
+                adaptor.getFunc(), adaptor.getLine(), rewriter);
+#else
     llAssert(op, condition, adaptor.getMessage(), adaptor.getFile(),
              adaptor.getFunc(), adaptor.getLine(), rewriter);
+#endif
     rewriter.eraseOp(op);
     return success();
   }
 
   // op: the op at which the assert is inserted. Unlike printf, we need to
   // know about the op to split the block.
+#ifdef USE_ROCM
+  void llAssertHIP(Operation *op, Value condition, StringRef message,
+                          StringRef file, StringRef func, int line,
+                          ConversionPatternRewriter &rewriter) const {
+    ConversionPatternRewriter::InsertionGuard guard(rewriter);
+    auto ctx = rewriter.getContext();
+    auto loc = op->getLoc();
+
+    // #block1
+    // if (condition) {
+    //   #block2
+    //   print(message);
+    //   halt;
+    // }
+    // #block3
```

Comment:
nitpick
```suggestion
    // #prebBlock
    // if (condition) {
    //   #ifBlock
    //   print(message);
    //   halt;
    // }
    // #thenBlock
```

Reply:
This is just optional suggestion, I think code is ok as-is.

Comment just makes understanding of code easier, so I suggest to use same names in both code and comment.

P.s. I have no good candidate for "thenBlock", so I can not propose anything about names.
It would be great if you find better names, though :)